code/modules/buildmode/submodes/lightmaker.dm

```diff
@@ -0,0 +1,31 @@
+/datum/buildmode_mode/lightmaker
+	key = "lightmaker"
+
+	var/light_range = 3
+	var/light_power = 1
+	var/light_color = COLOR_WHITE
+
+/datum/buildmode_mode/lightmaker/show_help(client/target_client)
+	to_chat(target_client, span_purple(examine_block(
+		"[span_bold("Left Click")] -> Create light\n\
+		[span_bold("Right Click")] -> Delete light\n\
+		[span_bold("Right Click on Build Mode Button")] -> Change light properties"))
+	)
+
+/datum/buildmode_mode/lightmaker/change_settings(client/target_client)
+	var/choice = alert("Change the new light range, power or color?", "Light Maker", "Range", "Power", "Color", "Cancel")
+	switch(choice)
+		if("Range")
+			light_range = input(target_client, "Range of light", text("Input")) as num|null
+		if("Power")
+			light_power = input(target_client, "Power of light", text("Input")) as num|null
+		if("Color")
+			light_color = input(target_client, "Light color", text("Input")) as color|null
+
+/datum/buildmode_mode/lightmaker/handle_click(client/target_client, params, obj/object)
+	var/list/modifiers = params2list(params)
+
+	if(LAZYACCESS(modifiers, LEFT_CLICK))
+		object.set_light(light_range, light_power, light_color)
+	if(LAZYACCESS(modifiers, RIGHT_CLICK))
+		object.set_light(0,0,0, COLOR_WHITE)
```

Comment:
That's not exactly deleting the light source

Reply:
> That's not exactly deleting the light source

Creating room with no light 
![image](https://github.com/shiptest-ss13/Shiptest/assets/105150564/bcc5bb27-6114-4d10-8a53-bba8b401b116)
VV: 
```
(E) (C) (M) light_color = "#FFFFFF"
(E) (C) (M) light_flags = 0
(E) (C) (M) light_on = 1
(E) (C) (M) light_power = 1
(E) (C) (M) light_range = 0
(E) (C) (M) light_sources = null
(E) (C) (M) light_system = 1
```
1. Deleting light by my code: 
```
(E) (C) (M) light_color = 0
(E) (C) (M) light_flags = 0
(E) (C) (M) light_on = "#FFFFFF" //now I see the error - it was working but still not accurate. 
(E) (C) (M) light_power = 0
(E) (C) (M) light_range = 0
(E) (C) (M) light_sources = null
(E) (C) (M) light_system = 1
```
2. Adding light with my code: 
```
(E) (C) (M) light_color = "#FFFFFF"
(E) (C) (M) light_flags = 0
(E) (C) (M) light_on = "#FFFFFF"
(E) (C) (M) light_power = 1
(E) (C) (M) light_range = 3
(E) (C) (M) light_sources = /list (1)
1 = /datum/light_source [0x210054c1]
(E) (C) (M) light_system = 1
```